objects/Stats.pm

```diff
@@ -1313,6 +1313,81 @@ sub statsList
         $this->{'total'} = sprintf "%.4f", $this->{'total_verticals'} / $this->{'total_firsts'};
       }
     },
+
+    {
+      Constants::STAT_NAME => 'Horizontal Openings per First',
+      Constants::STAT_ITEM_OBJECT_NAME => {Constants::STAT_OBJECT_DISPLAY_NAME => Constants::STAT_OBJECT_DISPLAY_PCAVG, 'total' => 0, 'total_firsts' => 0, 'total_verticals' => 0},
+      Constants::STAT_DATATYPE_NAME => Constants::DATATYPE_ITEM,
+      Constants::STAT_METATYPE_NAME => Constants::METATYPE_PLAYER,
+      Constants::STAT_COMBINE_FUNCTION_NAME =>
+      sub
+      {
+        my $this  = shift;
+        my $other = shift;
+        $this->{'total_horizontals'} += $other->{'total_horizontals'};
+        $this->{'total_firsts'}      += $other->{'total_firsts'};
+        if ($this->{'total_firsts'} == 0)
+        {
+          return;
+        }
+        $this->{'total'} = sprintf "%.4f", $this->{'total_horizontals'} / $this->{'total_firsts'};
+      },
+      Constants::STAT_ADD_FUNCTION_NAME =>
+      sub
+      {
+        my $this = shift;
+        my $game = shift;
+        my $this_player = shift;
+
+        $this->{'total_horizontals'}   += $game->getNumHorizontalOpeningPlays($this_player);
+        $this->{'total_firsts'}      += $game->getNumFirsts($this_player);
+	if ($this->{'total_firsts'} == 0)
+	{
+          return;
+	}
+        $this->{'total'} = sprintf "%.4f", $this->{'total_horizontals'} / $this->{'total_firsts'};
+      }
+    },
+
+
```

Comment:
The following stats seems to be an implementation of word type moves per first. I don't see any calls to get the number of exchanges.